drivers/media/platform/raspberrypi/pisp_be/pisp_be.c

```diff
@@ -600,53 +615,51 @@ static void pispbe_schedule(struct pispbe_dev *pispbe,
 	if (pispbe->hw_busy)
 		goto unlock_and_return;
 
-	for (unsigned int i = 0; i < PISPBE_NUM_NODE_GROUPS; i++) {
-		int ret;
+	job = list_first_entry_or_null(&pispbe->job_queue,
+				       struct pispbe_job_descriptor,
+				       queue);
+	if (!job)
+		goto unlock_and_return;
 
-		/* Schedule jobs only for a specific group. */
-		if (node_group && &pispbe->node_group[i] != node_group)
-			continue;
+	list_del(&job->queue);
 
-		/*
-		 * Prepare a job for this group, if the group is not ready
-		 * continue and try with the next one.
-		 */
-		ret = pispbe_prepare_job(&pispbe->node_group[i], &job);
-		if (ret)
-			continue;
+	for (unsigned int i = 0; i < PISPBE_NUM_NODES; i++)
+		pispbe->queued_job.buf[i] = job->buf[i];
+	pispbe->queued_job.node_group = job->node_group;
+
+	pispbe->hw_busy = true;
 
+	/*
+	 * We can kick the job off without the hw_lock, as this can
+	 * never run again until hw_busy is cleared, which will happen
+	 * only when the following job has been queued and an interrupt
+	 * is rised.
+	 */
+	spin_unlock_irqrestore(&pispbe->hw_lock, flags);
+
+	if (job->config->num_tiles <= 0 ||
```

Comment:
Is the above comment still relevant? (I must admit I only partially understood it before... but it seems unnecessary now if this function doesn't actually start the job).

Reply:
I think it still is relevant, as we're actually kicking off a job without the hw_lock being held, as the driver relies on the `hw_busy` flag to discern if the system is running a job or not ?